### PR TITLE
feat: introduce Text web component

### PR DESCRIPTION
**Pull request by ilhan007**
Introduce new web component.

Fixes: https://github.com/SAP/ui5-webcomponents/issues/6788

### API

```ts
/**
 * @class
 * @slot {Array<Node>} default - Defines the text of the component.
 * @since 1.23.0
 */
class Text extends UI5Element {
	/**
	 * Defines the number of lines the text should wrap before it truncates.
	 * @default Infinity
	 * @public
	 */
	@property({ validator: Integer, defaultValue: Infinity })
	maxLines!: number;
```
### Usage

```html
<ui5-text>Simple Text</ui5-text>
<ui5-text max-lines="1">Text that will truncate on single line</ui5-text>
<ui5-text max-lines="3">Text wraps to 3 lines and truncates afterwards</ui5-text>
```

Screenshot 2024-02-19 at 8 56 54

### CODE REVIEW

**Comment by MarcusNotheis** [comment]
From UI5 Web Components for React Side we would love to see a `Text` web component.
It would be great if you could support the following props as well because then we would be to remove our custom `Text` implementation in the React wrapper:
- [ ] `emptyIndicator` (SAPUI5 uses the `emptyIndicatorMode` for with the [EmptyIndicatorMode enum](https://ui5.sap.com/#/api/sap.m.EmptyIndicatorMode) that)
- [ ] `renderWhitespace`
- [ ] `hyphenated` (SAPUI5 uses `wrappingType=[WrappingType](https://ui5.sap.com/#/api/sap.m.WrappingType)` for that)

**Comment by petyabegovska** [comment]
Related to: #6788

File: packages/main/src/Text.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * @class
+ *
+ * <h3>Overview</h3>
+ *
+ * The Text component displays text that can be used in any content area of an application.
+ *
+ * <h3>Usage</h3>
+ *
+ * - Use the `Text` if uou want to display text inside a form, table, or any other content area.
+ * - Do not use the `Text` if you need to reference input type of components (use ui5-label).
```
Comment (BorisDafov) [inline packages/main/src/Text.ts:21]: I suggest:
- Do not use the `ui5-text` if you need to reference the input of other components (use `ui5-label` instead).

File: packages/main/src/Text.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+// Styles
+import styles from "./generated/themes/Text.css.js";
+
+/**
+ * @class
+ *
+ * <h3>Overview</h3>
+ *
+ * The Text component displays text that can be used in any content area of an application.
```
Comment (BorisDafov) [inline packages/main/src/Text.ts:16]: I suggest this formatting when referencing the component:
`ui5-text`

File: packages/main/src/Text.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * <h3>Usage</h3>
+ *
+ * - Use the `Text` if uou want to display text inside a form, table, or any other content area.
```
Comment (BorisDafov) [inline packages/main/src/Text.ts:20]: - Use the `ui5-text` if you ...

File: packages/main/src/Text.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ * - Do not use the `Text` if you need to reference input type of components (use ui5-label).
+ *
+ * <h3>Responsive behavior</h3>
+ *
+ * The Text component is fully adaptive to all screen sizes.
```
Comment (BorisDafov) [inline packages/main/src/Text.ts:25]: `ui5-text`

File: packages/main/test/pages/Text.html [hunk 5]
```diff
@@ -0,0 +1,65 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+	<meta charset="utf-8">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
+	<meta charset="utf-8">
+	<title>Text Web Component</title>
```
Comment (BorisDafov) [inline packages/main/test/pages/Text.html:8]: <title>Text</title>

**Review by BorisDafov — CHANGES_REQUESTED** [review]

**Comment by BorisDafov** [comment]
I approve the PR from UA perspective.

**Review by dimovpetar — APPROVED** [review]
Looks good 👍 
I will add test in another PR.

**Comment by ilhan007** [comment]
> From UI5 Web Components for React Side we would love to see a `Text` web component. It would be great if you could support the following props as well because then we would be to remove our custom `Text` implementation in the React wrapper:
> 
> * [ ]  `emptyIndicator` (SAPUI5 uses the `emptyIndicatorMode` with the [EmptyIndicatorMode enum](https://ui5.sap.com/#/api/sap.m.EmptyIndicatorMode) for that)
> * [ ]  `renderWhitespace`
> * [ ]  `hyphenated` (SAPUI5 uses `wrappingType` [WrappingType](https://ui5.sap.com/#/api/sap.m.WrappingType) for that)

Hi Marcus we had sync with design owners and all the features will be part of newest Text component design. Once the spec is finalised, we will implement them on top of the initial scope that we will merge now.

**Pull request merged**